From c19d24ab47674c2a82591c9e16fb450df7882465 Mon Sep 17 00:00:00 2001
From: Irine Sistiana <49315432+IrineSistiana@users.noreply.github.com>
Date: Thu, 21 Sep 2023 08:57:07 +0800
Subject: [PATCH 3/6] pool: fixed bytes pool size was 1k

---
 pkg/pool/allocator.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pkg/pool/allocator.go b/pkg/pool/allocator.go
index eb011ea..84b1110 100644
--- a/pkg/pool/allocator.go
+++ b/pkg/pool/allocator.go
@@ -24,7 +24,7 @@ import (
 )
 
 var (
-	_pool      = bytesPool.NewPool(10) // 1Mbyte pool, should be enough.
+	_pool      = bytesPool.NewPool(20) // 1Mbyte pool, should be enough.
 	GetBuf     = _pool.Get
 	ReleaseBuf = _pool.Release
 )
-- 
2.34.8

