From f0005ccc3a27dcbcc2266c550ffb7acf688523f0 Mon Sep 17 00:00:00 2001
From: Irine Sistiana <49315432+IrineSistiana@users.noreply.github.com>
Date: Thu, 21 Sep 2023 11:04:35 +0800
Subject: [PATCH 5/6] dnsutils: let WriteMsgToTCP use PackTCPBuffer

---
 pkg/dnsutils/net_io.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pkg/dnsutils/net_io.go b/pkg/dnsutils/net_io.go
index 8fc769b..f165446 100644
--- a/pkg/dnsutils/net_io.go
+++ b/pkg/dnsutils/net_io.go
@@ -78,12 +78,12 @@ func ReadMsgFromTCP(c io.Reader) (*dns.Msg, int, error) {
 // WriteMsgToTCP packs and writes m to c in RFC 1035 format.
 // n represents how many bytes are written to c.
 func WriteMsgToTCP(c io.Writer, m *dns.Msg) (n int, err error) {
-	mRaw, buf, err := pool.PackBuffer(m)
+	buf, err := pool.PackTCPBuffer(m)
 	if err != nil {
 		return 0, err
 	}
 	defer pool.ReleaseBuf(buf)
-	return WriteRawMsgToTCP(c, mRaw)
+	return c.Write(*buf)
 }
 
 // WriteRawMsgToTCP See WriteMsgToTCP
-- 
2.34.8

