--- a/client/src/login/Login/Form.js
+++ b/client/src/login/Login/Form.js
@@ -25,7 +25,7 @@ const Form = (props) => {
                         type="text"
                         className="form-control"
                         component={renderInputField}
-                        placeholder={t('username_placeholder')}
+                        placeholder="默认用户名密码都是root"
                         autoComplete="username"
                         autocapitalize="none"
                         disabled={processing}
